Source/WebCore/rendering/RenderFlexibleBox.cpp

```diff
@@ -1291,6 +1282,20 @@ LayoutUnit RenderFlexibleBox::computeFlexBaseSizeForFlexItem(RenderBox& flexItem
     return mainAxisExtent - mainAxisBorderAndPadding;
 }
 
+void RenderFlexibleBox::invalidateFlexItemsBeforeLayout()
+{
+    for (auto& flexItem : childrenOfType<RenderBox>(*this)) {
+        if (flexItem.needsLayout() || flexItem.isOutOfFlowPositioned() || flexItem.isExcludedFromNormalLayout())
+            continue;
+
+        if (flexItem.hasRelativeLogicalHeight()) {
+            auto computedLogicalHeight = flexItem.computeLogicalHeightWithoutLayout();
+            if (computedLogicalHeight != flexItem.logicalHeight())
+                flexItem.setNeedsLayout(MarkOnlyThis);
```

Comment:
I think this needs some comment (and the commit message should also be more specific). I am not sure how much we can trust in a precent (or calc) height value resolve right after calling  resetLogicalHeightBeforeLayoutIfNeeded(); on the containing block. Could you explain this a bit, please?